File: backend/src/main/java/com/api/flux/courseed/persistence/documents/Course.java
package com.api.flux.courseed.persistence.documents;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;

@Document(collection = "courses")
@Data
public class Course {
    @Id
    private String id;
    private String url;
    private String title;
    private String image;
    private String description;
    private String prerequisites;
    private Double price;
    private String duration;
    private String modality;
    private String categoryId;
    private String institutionId;
}

File: web-scraping/app/persistence/documents/Course.py
from mongoengine import Document, StringField, FloatField, IntField

class Course(Document):
    url = StringField(required=True, unique=True)
    title = StringField(required=True)
    image = StringField()
    originalImage = StringField()
    description = StringField()
    price = FloatField()
    duration = IntField()
    modality = StringField()
    type = StringField()
    categoryId = StringField(required=True)
    institutionId = StringField(required=True)

    meta = {
        "collection": "courses"
    }